rafiki/model/log.py

```diff
@@ -1,73 +1,190 @@
 import os
 import traceback
 import datetime
+import json
+import logging
 
-class DuplicatePlotException(Exception): pass
+MODEL_LOG_DATETIME_FORMAT = '%Y-%m-%dT%H:%M:%S'
 
-class ModelLogUtils():
+class LogType():
+    PLOT = 'PLOT'
+    METRICS = 'METRICS'
+    MESSAGE = 'MESSAGE'
+
+class ModelLogger():
     '''
-    Collection of utility methods for logging and plotting of messages & metrics during training.
-    '''   
-    def __init__(self):
-        # Add logging to stdout for local debugging
-        self._logger = ModelLogUtilsLogger()
+    Allows models to log messages and metrics during model training, and 
+    define plots for visualization of model training.
 
-    def set_logger(self, logger):
-        if not isinstance(logger, ModelLogUtilsLogger):
-            raise Exception('`logger` should subclass `ModelLogUtilsLogger`')
-        
-        self._logger = logger
+    To use this logger, import the global ``logger`` instance from the module ``rafiki.model``.
 
-    def log(self, message):
-        '''
-        Logs a message for analysis of model training.
-        '''
-        self._logger.log(message)
+    For example:
+
+    ::
+
+        from rafiki.model import logger, BaseModel
+        ...
+        class MyModel(BaseModel):
+            ...
+            def train(self, dataset_uri):
+                ...
+                logger.log('Starting model training...')
+                logger.define_plot('Precision & Recall', y_axis=['precision', 'recall'])
+                ...
+                logger.log(precision=0.1, recall=0.6, epoch=1)
```

Comment:
need to put epoch in `logger.define_plot('Precision & Recall', y_axis=['precision', 'recall'], x_axis='epoch')`